dipy/segment/tissue.py

```diff
@@ -5,57 +5,58 @@
-    Random Fields modeling approach
+    """
+    This class contains the methods for tissue classification using the
+    Markov Random Fields modeling approach.
     """
 
     def __init__(self, save_history=False, verbose=True):
-
         self.save_history = save_history
         self.segmentations = []
         self.pves = []
         self.energies = []
         self.energies_sum = []
         self.verbose = verbose
 
-    def classify(self, image, nclasses, beta, tolerance=None, max_iter=None):
-        r"""
+    def classify(self, image, nclasses, beta, tolerance=1e-05, max_iter=100):
+        """
         This method uses the Maximum a posteriori - Markov Random Field
-        approach for segmentation by using the Iterative Conditional Modes and
-        Expectation Maximization to estimate the parameters.
+        approach for segmentation by using the Iterative Conditional Modes
+        and Expectation Maximization to estimate the parameters.
 
         Parameters
         ----------
         image : ndarray,
-                3D structural image.
+            3D structural image.
         nclasses : int,
-                number of desired classes.
+            Number of desired classes.
         beta : float,
-                smoothing parameter, the higher this number the smoother the
-                output will be.
-        tolerance: float,
-                value that defines the percentage of change tolerated to
-                prevent the ICM loop to stop. Default is 1e-05.
-        max_iter : float,
-                fixed number of desired iterations. Default is 100.
-                If the user only specifies this parameter, the tolerance
-                value will not be considered. If none of these two
-                parameters
+            Smoothing parameter, the higher this number the smoother the
+            output will be.
+        tolerance: float, optional
+            Value that defines the percentage of change tolerated to
+            prevent the ICM loop to stop. Default is 1e-05.
+            If you want tolerance check to be disabled put 'tolerance = 0'.
+        max_iter : int, optional
+            Fixed number of desired iterations. Default is 100.
+            This parameter defines the maximum number of iterations the
+            algorithm will perform. The loop may terminate early if the
+            change in energy sum between iterations falls below the
+            threshold defined by `tolerance`. However, if `tolerance` is
+            explicitly set to 0, this early stopping mechanism is disabled,
+            and the algorithm will run for the specified number of
+            iterations unless another stopping criterion is met.
 
         Returns
         -------
         initial_segmentation : ndarray,
-                3D segmented image with all tissue types
-                specified in nclasses.
+            3D segmented image with all tissue types specified in nclasses.
```

Comment:
Style changes (and there are a lot here) should be on their own commit. Otherwise the review gets a lot harder, the real issue being addressed getting lost.

Reply:
Agree. We need to double check if this change follows the styles of other code files as well.